hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieCLIUtils.scala

```diff
@@ -109,10 +111,15 @@ object HoodieCLIUtils {
       .toMap
   }
 
-  def getLockOptions(tablePath: String): Map[String, String] = {
-    val props = FileSystemBasedLockProvider.getLockConfig(tablePath)
-    props.stringPropertyNames.asScala
-      .map(key => key -> props.getString(key))
-      .toMap
+  def getLockOptions(tablePath: String, schema: String, lockConfig: TypedProperties): Map[String, String] = {
+    val customSupportedFSs = lockConfig.getStringList(HoodieCommonConfig.HOODIE_FS_ATOMIC_CREATION_SUPPORT.key, ",", new ArrayList[String])
+    if (customSupportedFSs.contains(schema) || StorageSchemes.isAtomicCreationSupported(schema)) {
```

Comment:
Spark seems to have zookeeper as the default lock provider when multi-writer is needed, while Flink indeed configured a fs lock provider anyway, can you share the logic where fs lock provider is configured with spark ?

Reply:
An example: When using Spark to call the clustering procedure, there is a configuration to add an FS lock here.